File: Easy/2264. Largest 3-Same-Digit Number in String.java
class Solution {
    public String largestGoodInteger(String num) {
        String res = "";
        for (int i = 1; i < num.length() - 1; i++) {
            if (num.charAt(i - 1) == num.charAt(i) && num.charAt(i) == num.charAt(i + 1)) {
                if (res.length() == 0 || res.charAt(res.length() - 1) < num.charAt(i)) {
                    res = num.substring(i - 1, i + 2);
                }
            }
        }
        return res;
    }
}

File: Easy/2264. Largest 3-Same-Digit Number in String.py
class Solution:
    def largestGoodInteger(self, num: str) -> str:
        largest = -1
        curr_num = int(num[0])
        streak = 0
        for c in num:
            digit = int(c)
            if curr_num == digit:
                streak += 1
                if streak > 2:
                    largest = max(largest, digit)
            else:
                curr_num = digit
                streak = 1
        return "" if largest < 0 else str(largest) * 3